core/trino-main/src/main/java/io/trino/sql/planner/plan/TableWriterNode.java

```diff
@@ -245,21 +258,29 @@ public String toString()
         {
             return catalog + "." + tableMetadata.getTable();
         }
+
+        public String getCatalogName()
+        {
+            return catalog;
+        }
     }
 
     public static class CreateTarget
             extends WriterTarget
     {
         private final OutputTableHandle handle;
         private final SchemaTableName schemaTableName;
+        private final ConnectorTableMetadata tableMetadata;
 
         @JsonCreator
         public CreateTarget(
                 @JsonProperty("handle") OutputTableHandle handle,
-                @JsonProperty("schemaTableName") SchemaTableName schemaTableName)
+                @JsonProperty("schemaTableName") SchemaTableName schemaTableName,
+                ConnectorTableMetadata tableMetadata)
```

Comment:
This creates the know problem with serialization. I would recommend to have a `boolean reportingWrittenBytesSupported` field and set it based on the `CreateReference` in the `BeginTableWrite` rule.